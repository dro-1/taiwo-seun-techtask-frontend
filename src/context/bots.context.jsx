import React, { createContext, useState } from "react";

export const BotsContext = createContext({
  bots: null,
  setBots: () => null,
});

const BotsProvider = ({ children }) => {
  const [bots, setBots] = useState(null);

  return (
    <BotsContext.Provider
      value={{
        bots,
        setBots,
      }}
    >
      {children}
    </BotsContext.Provider>
  );
};
export default BotsProvider;
