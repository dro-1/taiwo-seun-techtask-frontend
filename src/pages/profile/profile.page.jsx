import { defaultProfile } from "../../services/data";
import "./profile.style.scss";
import userIcon from "./../../assets/svg/user.svg";
import sentIcon from "./../../assets/svg/send-msg.svg";
import receivingIcon from "./../../assets/svg/incoming-msg.svg";

export const Profile = () => {
  return (
    <div className="profile">
      <main>
        <img src={defaultProfile.image} alt="User" />
        <div className="info">
          <h1>{defaultProfile.name}</h1>
          <em>{defaultProfile.template}</em>
        </div>
        <p> Created: {new Date(defaultProfile.created).toDateString()}</p>
      </main>
      <div className="divider" />
      <div className="status">
        <h2>Status</h2>
        <em>Status: Free</em>
        <button>Activate</button>
      </div>
      <div className="divider" />
      <div className="extras">
        <div className="location">
          <div className="region">
            <strong>Region & Idiom</strong>
            <em>EUA - English(EN)</em>
          </div>
          <div className="time">
            <strong>Time Zone</strong>
            <em>EUA - English(EN)</em>
          </div>
        </div>
        <div className="user">
          <img src={userIcon} alt="" />
          <div className="info">
            <strong>{defaultProfile.analytics.user.actived}</strong>
            <em>Active Users</em>
          </div>
        </div>
        <div className="received-messages">
          <img src={receivingIcon} alt="" />
          <div className="info">
            <strong>{defaultProfile.analytics.message.received}</strong>
            <em>Received Messages</em>
          </div>
        </div>
        <div className="sent-messages">
          <img src={sentIcon} alt="" />
          <div className="info">
            <strong>{defaultProfile.analytics.message.sent}</strong>
            <em>Sent Messages</em>
          </div>
        </div>
      </div>
    </div>
  );
};
