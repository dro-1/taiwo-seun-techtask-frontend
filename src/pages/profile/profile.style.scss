div.profile {
  padding: 4rem 0;
  background: hsla(57, 98%, 50%, 1);

  background: linear-gradient(
    45deg,
    hsla(57, 98%, 50%, 1) 0%,
    hsla(325, 69%, 56%, 1) 100%
  );

  background: -moz-linear-gradient(
    45deg,
    hsla(57, 98%, 50%, 1) 0%,
    hsla(325, 69%, 56%, 1) 100%
  );

  background: -webkit-linear-gradient(
    45deg,
    hsla(57, 98%, 50%, 1) 0%,
    hsla(325, 69%, 56%, 1) 100%
  );
  main {
    width: 100%;
    padding: 1rem;

    margin: 2rem 0;
    color: #fff;
    img {
      width: 15rem;
      height: 15rem;
      display: block;
      margin-left: auto;
      margin-right: auto;
      margin-bottom: 1rem;
      border-radius: 50%;
    }
    div.info {
      text-align: center;
      h1 {
        font-size: 2rem;
      }
      em {
        font-size: 1.6rem;
      }
    }
    p {
      font-style: normal;
      font-size: 1.6rem;
      text-align: center;
    }
  }
  div.divider {
    width: 90%;
    height: 1px;
    background-color: #ddd;
    margin: 4rem auto;
  }
  div.status {
    text-align: center;
    color: #fff;
    margin-bottom: 2rem;
    h2 {
      font-size: 1.8rem;
      margin-bottom: 1rem;
    }
    em {
      font-style: normal;
      font-size: 1.6rem;
      display: block;
      margin-bottom: 1rem;
    }
    button {
      border: none;
      border-radius: 1rem;
      background-color: rgb(94, 94, 185);
      color: #fff;
      padding: 1rem;
    }
    div.divider {
      width: 90%;
      height: 1px;
      background-color: #ddd;
      margin: 4rem auto;
    }
  }
  div.location {
    background-color: #fff;
    border-radius: 1rem;
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 2rem;
    box-shadow: 0px 1px 4px 0px rgba(0, 0, 0, 0.75);
    padding: 1rem;
    div {
      margin-bottom: 1rem;
      strong {
        font-size: 1.6rem;
        font-style: normal;
        display: block;
      }
      em {
        font-size: 1.4rem;
        font-style: normal;
      }
    }
  }
  div.user,
  div.received-messages,
  div.sent-messages {
    background-color: #fff;
    border-radius: 1rem;
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 2rem;
    padding: 1rem;
    display: flex;
    align-items: center;
    box-shadow: 0px 1px 4px 0px rgba(0, 0, 0, 0.75);
    img {
      display: block;
      height: 5rem;
      width: 5rem;
      border-radius: 50%;
      margin-right: 2rem;
    }
    div {
      margin-bottom: 1rem;
      strong {
        font-size: 1.6rem;
        font-style: normal;
        display: block;
      }
      em {
        font-size: 1.4rem;
        font-style: normal;
      }
    }
  }

  div.received-messages,
  div.sent-messages {
    img {
      border-radius: 0;
    }
  }
}

@media screen and (min-width: 550px) {
  div.extras {
    display: grid;
    grid-template-columns: 1fr 1fr;
    max-width: 750px;
    margin-left: auto;
    margin-right: auto;
  }
}
