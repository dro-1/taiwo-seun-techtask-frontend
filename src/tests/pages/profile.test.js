import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "enzyme";
import { Profile } from "../../pages/profile/profile.page";
import { BotsContext } from "../../context/bots.context";
import { defaultProfile } from "./../../services/data";

describe("Profile Page", () => {
  it("correctly renders Profile Page Loader if bot is unavailable", () => {
    const wrapper = render(
      <BotsContext.Provider
        value={{
          getBot: () => null,
        }}
      >
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </BotsContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("correctly renders Bot Profile if bot is available", () => {
    const mockGetBot = jest.fn(() => {
      return defaultProfile;
    });
    const wrapper = render(
      <BotsContext.Provider
        value={{
          getBot: mockGetBot,
        }}
      >
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </BotsContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
