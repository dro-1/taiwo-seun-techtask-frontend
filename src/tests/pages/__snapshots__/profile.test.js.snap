// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page correctly renders Bot Profile if bot is available 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": " loader",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "loader-inner loader--style3",
          "title": "2",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "height": "100px",
              "id": "loader-1",
              "style": "enable-background:new 0 0 50 50",
              "version": "1.1",
              "viewBox": "0 0 50 50",
              "width": "100px",
              "x": "0px",
              "xmlns": "http://www.w3.org/2000/svg",
              "y": "0px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z",
                  "fill": "#000",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "attributeName": "transform",
                      "attributeType": "xml",
                      "dur": "0.6s",
                      "from": "0 25 25",
                      "repeatCount": "indefinite",
                      "to": "360 25 25",
                      "type": "rotate",
                    },
                    "children": Array [],
                    "name": "animateTransform",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "attributeName": undefined,
                      "attributeType": undefined,
                      "dur": undefined,
                      "from": undefined,
                      "repeatCount": undefined,
                      "to": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "attributeName": undefined,
                      "attributeType": undefined,
                      "dur": undefined,
                      "from": undefined,
                      "repeatCount": undefined,
                      "to": undefined,
                      "type": undefined,
                    },
                  },
                ],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "fill": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "fill": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "height": undefined,
              "id": undefined,
              "style": undefined,
              "version": undefined,
              "viewBox": undefined,
              "width": undefined,
              "x": undefined,
              "xmlns": "http://www.w3.org/2000/xmlns/",
              "y": undefined,
            },
            "x-attribsPrefix": Object {
              "height": undefined,
              "id": undefined,
              "style": undefined,
              "version": undefined,
              "viewBox": undefined,
              "width": undefined,
              "x": undefined,
              "xmlns": "",
              "y": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "title": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Profile Page correctly renders Profile Page Loader if bot is unavailable 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": " loader",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "loader-inner loader--style3",
          "title": "2",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "height": "100px",
              "id": "loader-1",
              "style": "enable-background:new 0 0 50 50",
              "version": "1.1",
              "viewBox": "0 0 50 50",
              "width": "100px",
              "x": "0px",
              "xmlns": "http://www.w3.org/2000/svg",
              "y": "0px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z",
                  "fill": "#000",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "attributeName": "transform",
                      "attributeType": "xml",
                      "dur": "0.6s",
                      "from": "0 25 25",
                      "repeatCount": "indefinite",
                      "to": "360 25 25",
                      "type": "rotate",
                    },
                    "children": Array [],
                    "name": "animateTransform",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "attributeName": undefined,
                      "attributeType": undefined,
                      "dur": undefined,
                      "from": undefined,
                      "repeatCount": undefined,
                      "to": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "attributeName": undefined,
                      "attributeType": undefined,
                      "dur": undefined,
                      "from": undefined,
                      "repeatCount": undefined,
                      "to": undefined,
                      "type": undefined,
                    },
                  },
                ],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "fill": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "fill": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "height": undefined,
              "id": undefined,
              "style": undefined,
              "version": undefined,
              "viewBox": undefined,
              "width": undefined,
              "x": undefined,
              "xmlns": "http://www.w3.org/2000/xmlns/",
              "y": undefined,
            },
            "x-attribsPrefix": Object {
              "height": undefined,
              "id": undefined,
              "style": undefined,
              "version": undefined,
              "viewBox": undefined,
              "width": undefined,
              "x": undefined,
              "xmlns": "",
              "y": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "title": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
